src/components/TodoContainer.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { Route, Switch } from 'react-router-dom';
+import Header from './Header';
+import InputTodo from './InputTodo';
+import TodosList from './TodosList';
+import About from '../pages/About';
+import NotMatch from '../pages/NotMatch';
+import Navbar from './Navbar';
+
+const TodoContainer = () => {
+  const initialValue = [
+    {
+      id: uuidv4(),
+      title: 'First sample',
+      completed: false,
+    },
+  ];
+  const [todos, setTodos] = useState(
+    localStorage.todos ? JSON.parse(localStorage.todos) : initialValue,
+  );
+
+  const updateLocalStorage = () => {
+    localStorage.todos = JSON.stringify(todos);
+  };
+  updateLocalStorage();
+
+  const handleChange = (id) => {
+    setTodos((prevState) => prevState.map(
+      (todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+        return todo;
+      },
+    ));
+    updateLocalStorage();
+  };
+
+  const delTodo = (id) => {
+    setTodos([
+      ...todos.filter((todo) => todo.id !== id),
+    ]);
```

Comment:
Please, prefer using the same method for using state variables in their own setter methods (say, calls like `setStateVar(... stateVar ...)`). If the callback is needed because of the asynchronicity of the update this use could lead to hard-to-trace errors. Otherwise, the callback is not needed in the snippet of the previous comment.